components/src/store/state.d.ts

```diff
@@ -9,6 +9,11 @@ export interface PreferencesState {
     cloud: CloudKey,
 }
 
+export interface FormState {
+    submittedFormIds: string[];
+}
+
 export interface AppState {
-    preferences: PreferencesState
+    preferences: PreferencesState;
+    form: FormState;
```

Comment:
What about something at the top level like `history`, with form submissions contained within it?